internal/stores/stores.go

```diff
@@ -39,3 +44,41 @@ func openOrCreateFile(fsys afero.Fs, filename string) (afero.File, error) {
 func inferObjPath(cfilePath string) string {
 	return strings.TrimSuffix(cfilePath, filepath.Ext(cfilePath))
 }
+
+// WriteMetadata generates metadata for
+func WriteMetadataToFsys(s Store, fullObjectPath string, f afero.File) (cleanup func() error, err error) {
+	opts := s.GetOptions()
+	if opts.MetadataFileExtension == "" {
+		return nil, metadata.ErrFileExtensionEmpty
+	}
+	fsys := s.GetFsys()
+	logger.V(2).Infof("object: %s", fullObjectPath)
+
+	// generate metadata
+	m, err := metadata.GenerateFromFile(f)
+	if err != nil {
+		return nil, err
+	}
+	logger.V(2).Infof("%s has a checksum of %q", fullObjectPath, m.Checksum)
+	// convert metadata to json
+	blob, err := json.MarshalIndent(m, "", " ")
+	if err != nil {
+		return nil, err
+	}
+	// Create path for metadata if it doesn't already exist
+	if err := fsys.MkdirAll(filepath.Dir(filepath.Dir(fullObjectPath)), os.ModePerm); err != nil {
+		return nil, err
+	}
```

Comment:
Its impossible for this metadata path to not exist as we are reading a file from the same path. We should remove this check.

The other edge case is that we are trying to write to a different path than that of the file we read to generate this metadata. If that happens, we want it to error out on line 75.